src/jit/codegenlegacy.cpp

```diff
@@ -19047,16 +19047,32 @@ regMaskTP CodeGen::genCodeForCall(GenTreeCall* call, bool valUsed)
                     {
                         noway_assert(callType == CT_USER_FUNC);
 
-                        void* pAddr;
-                        addr = compiler->info.compCompHnd->getAddressOfPInvokeFixup(methHnd, (void**)&pAddr);
-                        if (addr != NULL)
+                        CORINFO_CONST_LOOKUP lookup;
+                        compiler->info.compCompHnd->getAddressOfPInvokeTarget(methHnd, &lookup);
+
+                        void* addr = lookup.addr;
+
+                        _ASSERTE(addr != NULL);
+
+                        bool direct;
+#if defined(_TARGET_ARM_)
+                        if (lookup.accessType == IAT_VALUE)
```

Comment:
You can move this into the `noway_assert` and get rid of the `direct` local variable.